### PR TITLE
Font Size Picker: Add default size

### LINKED ISSUE #17908: Font Size Picker inconsistencies--is "normal" a required size?
**Describe the bug**
If a theme declares custom `editor-font-sizes`, but does not include a `normal` font size, the font picker selection ends up with some unexpected behaviors, such as:
* the first defined font size is set as the "default", but not applied
* in order to set the first defined font size as the size for the block, the user needs to select a different font size, and then select the first one
* looking at the font size under Text Settings, there is confusion because it may be showing "Small" as the font size, even if that's not what's been applied.

Looking at the FontSizePicker component, it appears that the "default" or fallback font size is `normal` ([ref](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/font-size-picker/index.js#L20)), which results in this unexpected behavior when there is no defined `normal` font size.

**To reproduce**
1. Activate the Twenty Twenty theme
2. Create a new post and add some content (paragraphs)
3. Look at the Text Settings panel on the side
4. See "Small" for Font Size, which suggests that the current paragraph is set to the Small font size, when actually no size has been applied

**Expected behavior**
When a paragraph or other block allows for a font size picker, I would expect an empty option to be the default state of the select field, especially if not changing the select field results in no class or custom style being applied.

Ideally, I would like the first/default option to be labeled "Default", with no value, which seems consistent with what's being saved to the block content (currently no value if nothing is changed or if the font size is reset).

However, if a specific font size (normal) is required, I think that information needs to be added to the [Block Editor Handbook](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#block-font-sizes).

Themes can "fix" this by always defining a `normal` font size, although even that can have some possible issues: for example, Twenty Twenty (in 5.3 nightly) sets the body font size to `1.8rem`, so if a user selects the "Regular" font size from the font picker dropdown, the block is actually rendered at 16px because of how the `has-regular-font-size` class is defined in the block-library/style.css, and the theme does not currently redefine that font size.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Firefox, Chrome
 - Version: FF/69.0.2 (64-bit), Chrome/77.0.3865.90 (Official Build) (64-bit) 

**Additional context**
- Gutenberg 6.6 and/or WP 5.3 nightly

### PR DESCRIPTION
**Pull request by mmtr**
## Description

Adds a default option to the font size picker component that unsets any font size.

Previously, the font size selector was falling back to the `normal` size, but this size might have not been defined by theme.

Fixes #17908

## How has this been tested?
- Activated the ~Twenty Twenty theme~ or any other theme providing font sizes not including a `normal` size (Twenty Twenty recently added a `normal` font size: https://github.com/WordPress/twentytwenty/pull/989).
- Created a new post and added some paragraph blocks.
- Made sure the Text Settings on the right sidebar has the "Default" font size, with no font size applied to the block.
- Tried setting any other font size and made sure it worked as expected.
- Clicked on "Reset" and verified the selector changed to "Default", with no font size applied to the block.
- Removed the font size number (after toggling any font size) and verified the selector changed to "Default", with no font size applied to the block.

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by kwight** [comment]
Works great 👍

File: docs/designers-developers/developers/themes/theme-support.md [hunk 1]
```diff
@@ -216,6 +216,8 @@ As an example for the regular font size, a theme may provide the following class
 }
 ```
 
+**Note:** The slugs `default` and `custom` are reserved and cannot be used by themes.
```
Comment (jorgefilipecosta) [inline docs/designers-developers/developers/themes/theme-support.md]: I guess we should also document this in https://github.com/WordPress/gutenberg//blob/6d4382efa125d385e57bfc835057bd0c44a51347/docs/designers-developers/developers/themes/theme-support.md#L177.

File: packages/components/src/font-size-picker/index.js [hunk 2]
```diff
@@ -12,18 +12,22 @@ import Button from '../button';
...
-	return 'normal';
+	return DEFAULT_FONT_SIZE;
 }
 
 function getSelectOptions( optionsArray ) {
 	return [
+		{ value: DEFAULT_FONT_SIZE, label: __( 'Default' ) },
 		...optionsArray.map( ( option ) => ( { value: option.slug, label: option.name } ) ),
-		{ value: 'custom', label: __( 'Custom' ) },
+		{ value: CUSTOM_FONT_SIZE, label: __( 'Custom' ) },
```
Comment (jorgefilipecosta) [inline packages/components/src/font-size-picker/index.js]: If I select a font size in the select control and then I select the custom options, I get some JavaScript errors:
```
react-dom.8b3dda97.js?ver=16.9.0:481 Uncaught TypeError: Cannot read property 'size' of undefined
    at onSelectChangeValue (index.js:65)
    at onChangeValue (index.js:35)
    at HTMLUnknownElement.callCallback (react-dom.8b3dda97.js?ver=16.9.0:341)
    at Object.invokeGuardedCallbackDev (react-dom.8b3dda97.js?ver=16.9.0:391)
    at invokeGuardedCallback (react-dom.8b3dda97.js?ver=16.9.0:448)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.8b3dda97.js?ver=16.9.0:462)
    at executeDispatch (react-dom.8b3dda97.js?ver=16.9.0:594)
    at executeDispatchesInOrder (react-dom.8b3dda97.js?ver=16.9.0:616)
    at executeDispatchesAndRelease (react-dom.8b3dda97.js?ver=16.9.0:719)
    at executeDispatchesAndReleaseTopLevel (react-dom.8b3dda97.js?ver=16.9.0:727)
```

**Review by jorgefilipecosta — COMMENTED** [review]
I left some comments, but the general idea looks good to me :)

File: docs/designers-developers/developers/themes/theme-support.md (hunk 1 above)
Comment (mmtr) [reply docs/designers-developers/developers/themes/theme-support.md]: Is not this PR already documenting it on that section or am I missing something?

File: packages/components/src/font-size-picker/index.js (hunk 2 above)
Comment (mmtr) [reply packages/components/src/font-size-picker/index.js]: It should have been fixed on c27a7409b47bd370c229c4c5bef0eec7c8050d7c

File: docs/designers-developers/developers/themes/theme-support.md (hunk 1 above)
Comment (jorgefilipecosta) [reply docs/designers-developers/developers/themes/theme-support.md]: I'm sorry I posted the wrong link. I wanted to refer the readme of the component `packages/components/src/font-size-picker/README.md`. The component also receives slugs and these slugs should also not be the reserved ones.

**Review by jorgefilipecosta — APPROVED** [review]
Works well on my tests 👍

File: docs/designers-developers/developers/themes/theme-support.md (hunk 1 above)
Comment (mmtr) [reply docs/designers-developers/developers/themes/theme-support.md]: Gotcha. Added there too 15f0fd9

**Pull request merged**